### PR TITLE
[CM-943] - Added support for GIFs

### PR DESCRIPTION
**Pull request by amntoppo**
## What do you want to achieve?
- GIFs were not handled by the SDK. Every image was being shown as a Bitmap image, hence GIFs were not animating.
- Added a check to see if Image coming is a GIF. If it's a GIF, load it in a different image viewer as an animating drawable.
- Zooming in is not possible for GIF images.
- Images can come from 2 places: Normal upload and Rich Message payload. Added support for GIF in both.

### CODE REVIEW

File: kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/activity/FullScreenImageActivity.java [hunk 1]
```diff
@@ -69,28 +72,42 @@ protected void onCreate(Bundle savedInstanceState) {
...
-                            mediaImageView.setImageBitmap(resource);
-                            mediaImageView.setZoom(1);
-                        }
-                    });
-
+            if(payloadModel.getUrl().endsWith("gif")) {
+                Glide.with(this)
+                        .asGif()
+                        .load(payloadModel.getUrl())
+                        .apply(new RequestOptions().override(1600, 1600)) //This is important
```
Comment (adarshmishra) [inline kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/activity/FullScreenImageActivity.java:87]: What is this 1600 hardcoded value.

Comment (amntoppo) [reply kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/activity/FullScreenImageActivity.java:87]: Those are width and height of the image

Comment (adarshmishra) [reply kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/activity/FullScreenImageActivity.java:87]: @amntoppo please move it to constant.

File: kommunicateui/src/main/res/layout/mobicom_image_full_screen.xml [hunk 2]
```diff
@@ -25,6 +32,9 @@
         android:adjustViewBounds="true"
         android:scaleType="fitCenter" />
 
+
+
+
```
Comment (Sathyan-Elangovan) [inline kommunicateui/src/main/res/layout/mobicom_image_full_screen.xml:37]: Remove extra lines

File: kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/activity/FullScreenImageActivity.java [hunk 3]
```diff
@@ -104,8 +123,16 @@ public void onResourceReady(@NonNull Bitmap resource, @Nullable Transition<? sup
...
-                    Bitmap imageBitmap = ImageUtils.decodeSampledBitmapFromPath(message.getFilePaths().get(0));
-                    mediaImageView.setImageBitmap(imageBitmap);
+                    if(message.getFileMetas().getContentType().contains("gif")) {
+                        Glide.with(this)
+                                .asGif()
+                                .load(message.getFilePaths().get(0))
+                                .into(gifImageView);
+                    }
+                    else {
+                        Bitmap imageBitmap = ImageUtils.decodeSampledBitmapFromPath(message.getFilePaths().get(0));
```
Comment (Sathyan-Elangovan) [inline kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/activity/FullScreenImageActivity.java:133]: Are we checking this getFilePaths().length anywhere? in this case if message.getFilePaths() is empty , this line will produce crash

File: kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/activity/FullScreenImageActivity.java [hunk 4]
```diff
@@ -104,8 +123,16 @@ public void onResourceReady(@NonNull Bitmap resource, @Nullable Transition<? sup
 
             if (message != null && message.getFilePaths() != null && !message.getFilePaths().isEmpty()) {
                 try {
-                    Bitmap imageBitmap = ImageUtils.decodeSampledBitmapFromPath(message.getFilePaths().get(0));
-                    mediaImageView.setImageBitmap(imageBitmap);
+                    if(message.getFileMetas().getContentType().contains("gif")) {
+                        Glide.with(this)
+                                .asGif()
+                                .load(message.getFilePaths().get(0))
```
Comment (Sathyan-Elangovan) [inline kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/activity/FullScreenImageActivity.java:129]: same here get(0) will produce fatal error. Check the next comment

**Review by Sathyan-Elangovan — CHANGES_REQUESTED** [review]

File: kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/activity/FullScreenImageActivity.java (hunk 4 above)
Comment (amntoppo) [reply kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/activity/FullScreenImageActivity.java:129]: We are already doing a check:
if (message != null && message.getFilePaths() != null && !message.getFilePaths().isEmpty())

File: kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/activity/FullScreenImageActivity.java (hunk 3 above)
Comment (amntoppo) [reply kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/activity/FullScreenImageActivity.java:133]: We are already doing a check:
if (message != null && message.getFilePaths() != null && !message.getFilePaths().isEmpty())

**Review by Sathyan-Elangovan — APPROVED** [review]

**Review by adarshmishra — APPROVED** [review]

**Pull request merged**